streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -153,6 +153,27 @@
      */
     public static final String CONSUMER_PREFIX = "consumer.";
 
+    /**
+     * Prefix used to isolate {@link KafkaConsumer main consumer} configs from other client configs.
```

Comment:
nit: `isolate ... form other client configs` -> 

```
override ... for the main consumer client from the general consumer client configs. The override precedence is the following (from highest to lowest precedence):

1. main.consumer.[config-name]
2. consumer.[config-name]
3. [config-name]
```

Ditto below for other two.